### PR TITLE
Multitenancy import

### PR DESCRIPTION
**Pull request by ksamoray**
Implement import functionality for supported resources

### CODE REVIEW

File: nsxt/resource_nsxt_policy_tier1_gateway_interface.go [hunk 1]
```diff
@@ -316,14 +317,21 @@ func resourceNsxtPolicyTier1GatewayInterfaceDelete(d *schema.ResourceData, m int
 func resourceNsxtPolicyTier1GatewayInterfaceImport(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
 	importID := d.Id()
 	s := strings.Split(importID, "/")
```
Comment (annakhm) [inline nsxt/resource_nsxt_policy_tier1_gateway_interface.go:319]: cosmetics: lets move this below the path import section?

Comment (ksamoray) [reply nsxt/resource_nsxt_policy_tier1_gateway_interface.go:319]: The split? It's used at line 323 to get the locale_service_id and then again in 328. So IDK if there's a better place for it.
Unless you mean something else.

Comment (annakhm) [reply nsxt/resource_nsxt_policy_tier1_gateway_interface.go:319]: Correct, my bad, sorry. But it seems wrong to reference s[9] below? Doesn't index depend on whether tenant is present in path? And the user can provide path with smaller amount of tokens, that would still be policy path

File: nsxt/policy_utils.go [hunk 2]
```diff
@@ -180,6 +196,39 @@ func nsxtDomainResourceImporter(d *schema.ResourceData, m interface{}) ([]*schem
...
+		return rd, err
+	}
+	return rd, nil
+}
+
+func nsxtPolicyPathResourceImporterHelper(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	importID := d.Id()
+	if isPolicyPath(importID) {
+		pathSegs := strings.Split(importID, "/")
+		if len(pathSegs) < 4 {
```
Comment (annakhm) [inline nsxt/policy_utils.go:213]: This is already validated in isPolicyPath

File: nsxt/policy_utils.go [hunk 3]
```diff
@@ -180,6 +196,39 @@ func nsxtDomainResourceImporter(d *schema.ResourceData, m interface{}) ([]*schem
...
+			return nil, fmt.Errorf("invalid policy path %s", importID)
+		} else if strings.Contains(pathSegs[1], "infra") {
+			d.SetId(pathSegs[len(pathSegs)-1])
+		} else if pathSegs[1] == "orgs" && pathSegs[3] == "projects" {
+			// pathSegs[2] should contain the organization. Once we support multiple organization, it should be
+			// assigned into the context as well
+			contexts := make([]interface{}, 1)
+			ctxMap := make(map[string]interface{})
+			ctxMap["project_id"] = pathSegs[4]
```
Comment (annakhm) [inline nsxt/policy_utils.go:222]: can we get a segmentation fault here if user provides bad path?

File: nsxt/resource_nsxt_policy_tier1_gateway_interface.go (hunk 1 above)
Comment (ksamoray) [reply nsxt/resource_nsxt_policy_tier1_gateway_interface.go:319]: Indeed. It's better to locate "locale_service" and grab the next list member.

File: nsxt/policy_utils.go (hunk 3 above)
Comment (ksamoray) [reply nsxt/policy_utils.go:222]: So maybe we should check for len(pathSegs < 5) for project case

**Review by annakhm — APPROVED** [review]

**Pull request merged**